### ISSUE TITLE
Support for importing Wigle Wifi GSM cell tower databases into AIMSICD 

### ISSUE THREAD

**Issue by ProGamerGov** [SecUpwN/Android-IMSI-Catcher-Detector#751]
I am sure you already know about Wigle Wifi as the developers originally closed another "feature request/suggestion" from another user here for Wifi integration (this post is purely about the cellular applications of the app), that was related to it here: https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/545

So first off whenever I have GPS issues with AIMSICD, the Wigle WIfi Android app always seems to fix the GPS. I like to run it alongside AIMSICD in order to capture the nearby wifi networks, in an attempt to find rogue/evil access points.  On my Wigle Wifi app database, I have hundreds of cell towers recorded multiple times, and some cell towers with thousands of data entries. 

The Wigle Wifi app records a massive amount of GSM data in addition to Wifi access points. The app allows you to export KML and CSV files for the entire data base, or just the current "run/session". The CSV file contains multiple different columns:

"MAC", "SSID", "AuthMode", "FirstSeen", "Channel", "RSSI", "CurrentLatitude", "CurrentLongitude", "AltitudeMeters", "AccuracyMeters", "Type".

Under "Type", the cell tower or wifi access point is labled as "Wifi" or "GSM". In Microsoft Excel, I am able to remove all the "wifi" data entries, so that I am left with the cell tower related entries. 

"SSID": The name of the cell company. Bell, Telus, Orange F, F-Bouygues Telecom, etc...

"MAC" : The number that contains country code, carrier, etc... AIMSICD displays it on the phone/SIM details page as the Operator ID, LAC, and CID.  

"Authmode": Shows the type of connection used, like "HSPA;fr", "UMTS;fr", "EDGE;fr", "HSPA;ca", "UMTS:de", "GPRS;fr", etc...

"FirstSeen": example: "2016-08-20 9:18:00 AM".

"Channel": Has no use with cell tower data, the value is zero on any data entry with "Type" set to "GSM". 

"RSSI": The signal strength.

The rest are self explanatory. 

So basically the Wigle Wifi App is constantly collecting a lot of the same data that AIMSICD collects for cell towers. It would be nice to utilize this information collected by Wigle Wifi, in AIMSICD as an alternative data base, or even as part of the main database. 

For importing the data from the Wigle Wifi CSV file, you would first have to remove the Wifi data via only copying the GSM type. The location, time, etc.. values can then been pulled and organized into AIMSICD's database. 

The Wigle Wifi app for Android can be found here: https://play.google.com/store/apps/details?id=net.wigle.wigleandroid&hl=en

The Wigle Wifi site can be found here: https://wigle.net/

**Comment by SecUpwN** [comment]
Thanks for proposing thi, @ProGamerGov. At the moment we simply have way too many Issues and do not see the need for adding this, nor the enhancement that it would provide. Pull requests are welcome.

**Comment by ProGamerGov** [comment]
Thinking farther towards maybe doing this myself...


Wigle = AIMSICD

WigleWifi_20160113201516.csv = aimsicd-DBi_bts.csv


Last AIMSICD data entry number  + 1 =  id

The "MAC" value can be separated into 4 separate values with a single "0" discarded. 

MCC MNC0_LAC_CID = MCC + MNC + LAC + CID

CurrentLatitude = gps_lat

CurrentLongitude = gps_lon 

00000000 = 0000

10 digits = 4 digits

Is it in the interest to round the number to a smaller value, or keep it's size?



FirstSeen = time_first and/or time_last


Not sure what the time format conversion is?

2015-08-15  6:15:00 PM = 20151009092501

Not sure where the RSSI value would be put either?

RSSI = negative 3 digit number = somewhere in AIMSICD?

AccuracyMeters = Anything in AIMSICD?

Its 1 to 3 digits in length.

Is the AuthMode value needed in AIMSICD?

The SSID value is the operator's name, which can be determined from the MNC value. So the SSID value can be discarded. 

I do not think Wigle has the PSC value, so it can just be set to zero for each entry.

**Comment by SecUpwN** [comment]
>Thinking farther towards maybe doing this myself...

Thanks for considering to craft a pull request, @ProGamerGov.

>Is it in the interest to round the number to a smaller value, or keep it's size?

Not sure. @larsgrefer, what would be the best way for moving towards #655?

>RSSI = negative 3 digit number = somewhere in AIMSICD?
>AccuracyMeters = Anything in AIMSICD?

Regarding these questions, I would like @He3556 to chime in with his knowledge.

>Also, would a simple manual program for adding the Wigle data onto the AIMSICD database, be ok for now? Created in something like Visual Basic?

If you open the [Navuigation Drawer](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Navigation-Drawer), you see the submenu `SETTINGS`. It would be great if another button `Import Database` could be added there, which would do the job you described here. Alternatively, we could consider expanding the already existing button `Restore Database`, which then could detect the Wigle Wifi Database if placed into `/sdcard/Android/data/com.SecUpwN.AIMSICD/files/WiggleWifi`.

**Comment by ProGamerGov** [comment]
For the map viewer, I am not sure how adding 200,000 measurements would affect the map viewer's stability. Is there anything in place at the moment for this potential issue?

**Comment by SecUpwN** [comment]
>For the map viewer, I am not sure how adding 200,000 measurements would affect the map viewer's stability.

I guess we'd have to implement the ability to import the database and simple see what happens.

>Is there anything in place at the moment for this potential issue?

Likely not. As said, we're already coping with all the existing Issues and too few Android developers.
